### ISSUE TITLE
[Issue] azd CLI is not detected  in VS Code terminal  

### ISSUE THREAD

**Issue by indcoder** [Azure/azure-dev#1441]
- [x] Make sure you've installed the latest version using [instructions in the wiki](../wiki/install)

**Output from `azd version`**
azd version 0.5.0-beta.3 (commit 1c23678aade587d5b9685a256a3c1d2d7f4b42e0)

**Output from `az version`** 
Run `az version` and copy and paste the output here (minimum required version is 2.38.0): 
{
  "azure-cli": "2.43.0",
  "azure-cli-core": "2.43.0",
  "azure-cli-telemetry": "1.0.8",
  "extensions": {}
}

**Describe the bug**
While the azd cli can be run from the Windows terminal across different shells,  

<img width="863" alt="image" src="https://user-images.githubusercontent.com/2099300/213986624-38b2d409-fd8c-49c2-8793-0d6bb210309c.png">

it fails when done via VS Code terminal

<img width="629" alt="image" src="https://user-images.githubusercontent.com/2099300/213987204-7289186c-7f51-4e7f-8f17-5d8685c39303.png">

I presume this is the root cause for the failure of  azd init when executed via VS Code Command Palette

<img width="758" alt="image" src="https://user-images.githubusercontent.com/2099300/213989781-2f66c015-14f0-428f-9449-82c557b87eb7.png">

Apparently it cannot detect azd.

<img width="634" alt="image" src="https://user-images.githubusercontent.com/2099300/213990073-0dcdc777-47b9-478e-af84-5fd13c9b9bcd.png">

**To Reproduce**
Run the azd -h command in the VS Code terminal
or
Any azd command via the VS Code  command palette

**Expected behavior**
The azd init command should be run and the project files generated.

**Environment**
Information on your environment:
    * Language name and version
    * IDE and version : [e.g. Visual Studio 16.3] 
       VS Code Insider 
       Version: 1.75.0-insider (user setup)
        Commit: 1379f0375440df946adc62f7afb8375dd45d5adf
        Date: 2023-01-17T05:21:07.217Z
        Electron: 19.1.9
        Chromium: 102.0.5005.194
        Node.js: 16.14.2
        V8: 10.2.154.23-electron.0
        OS: Windows_NT x64 10.0.22623
        Sandboxed: Yes    

**Additional context**
I'm presuming the azd-cli is not being added to the path which can be detected by VS Code terminal when installation is done via MSI
azure-cli-2.43.0.msi

**Comment by bwateratmsft** [comment]
This is similar to https://github.com/microsoft/vscode-docker/issues/3806. I wonder if there's a VSCode issue with discovery of things on the `PATH`.

**Comment by bwateratmsft** [comment]
@indcoder In a Nushell terminal window in VSCode, can you check the value of the `PATH` environment variable? I believe the command is `$env.PATH`. Does that value contain `C:\Users\YourUserName\AppData\Local\Programs\Azure Dev CLI\`?

**Comment by indcoder** [comment]
Yes, the path environment variable has that value 

But, I don't think so it has to do something with nu....because the Command palette is executing in its own task process[most probably powershell....looking the at the error message]

<img width="880" alt="image" src="https://user-images.githubusercontent.com/2099300/214094819-3e490b67-0011-4fd3-a495-7d0e3dcb58c2.png">

**Comment by bwateratmsft** [comment]
@indcoder does the issue occur with Electron sandboxing disabled?
![image](https://user-images.githubusercontent.com/36966225/214106982-2d093ac1-3a6b-4817-a24d-804d067e3616.png)

**Comment by jongio** [comment]
Did you completely close VS Code after installing azd?

**Comment by indcoder** [comment]
Yes, multiple times. I have opened this issue after doing all these steps,
multiple times.


Aslo azd works in the Windows terminal opened independently. Its only in
the context of VS Code, I'm getting to see this behaviour.

**Comment by bwateratmsft** [comment]
This may unblock you, but there is an environment variable, `AZURE_DEV_CLI_PATH`, that can be used to set the path to the Azure Dev CLI. Generally it's located at `C:\Users\YourUserName\AppData\Local\Programs\Azure Dev CLI\azd.exe`.

It seems like some part of the `PATH` isn't making it into shell processes spawned by the extension host or VSCode itself (which would include terminal windows in VSCode), even though VSCode *does* have the right values in `process.env.PATH`. I tried with sandboxing both enabled and disabled and everything worked. I wonder if there's some Node.js setting causing this.

**Comment by bwateratmsft** [comment]
A Docker extension user filed https://github.com/microsoft/vscode/issues/172124, which might be the root cause of this issue and the issues that user is seeing.

**Comment by savannahostrowski** [comment]
> I'm presuming the azd-cli is not being added to the path which can be detected by VS Code terminal when installation is done via MSI - azure-cli-2.43.0.msi

@indcoder Can you confirm that you've installed azd from aka.ms/azd-install? Looks like you're referring to the Azure CLI above (which is a different tool)!

**Comment by indcoder** [comment]
Yes, Sarah I have installed from that location. Also as mentioned in my original post, the issue is not in azd cli where it works, but in the azd vs code extension.

**Comment by bwateratmsft** [comment]
Aside from something upstream like https://github.com/microsoft/vscode/issues/172124 I'm not sure what could be happening. `PATH` seems all messed up lately in VSCode.

**Comment by rajeshkamal5050** [comment]
@bwateratmsft want to keep this open or close it as not reproducible?

**Comment by bwateratmsft** [comment]
@rajeshkamal5050 I think we can close it unless we can find more information.

**Comment by indcoder** [comment]
So I was still getting this issue  on my machine after I did a complete reinstall of the OS due to a hardware issue..

A bit of digging I keep came across a stack overflow issue for other cmdlets not being found ....a solution was proposed regarding VS code attaching the incorrent Powershell...so from the VS Code command palette 

<img width="861" alt="image" src="https://github.com/Azure/azure-dev/assets/2099300/8a692e19-74f4-4dc5-a572-e6a901ea18f9">

Ensure you pick the correct version of powershell 

<img width="905" alt="image" src="https://github.com/Azure/azure-dev/assets/2099300/a9dbca58-1b4f-4100-8e8b-4d597a546372">

and then the azd cli extension worked correctly and the template dropdown was available

Putting this out here if some one also faces this issue.
